binr/r2r/r2r.c

```diff
@@ -41,23 +41,78 @@ static int help(bool verbose) {
 		" -r [radare2] path to radare2 executable (default is "RADARE2_CMD_DEFAULT")\n"
 		" -m [rasm2]   path to rasm2 executable (default is "RASM2_CMD_DEFAULT")\n"
 		" -f [file]    file to use for json tests (default is "JSON_TEST_FILE_DEFAULT")\n"
+		" -C [dir]     chdir before running r2r (default follows executable symlink + test/new\n"
 		"\n"
 		"OS/Arch for archos tests: "R2R_ARCH_OS"\n");
 	}
 	return 1;
 }
 
+static bool r2r_chdir(const char *argv0) {
+#if __UNIX__
+	if (r_file_is_directory ("db")) {
+		return true;
+	}
+	char src_path[PATH_MAX];
+	char *r2r_path = r_file_path (argv0);
+	bool found = false;
+	if (readlink (r2r_path, src_path, sizeof (src_path)) != -1) {
+		char *p = strstr (src_path, R_SYS_DIR "binr"R_SYS_DIR"r2r"R_SYS_DIR"r2r");
+		if (p) {
+			*p = 0;
+			strcat (src_path, R_SYS_DIR"test"R_SYS_DIR"new");
+			if (r_file_is_directory (src_path)) {
+				(void)chdir (src_path);
+				eprintf ("Running from %s\n", src_path);
+				found = true;
+			}
+		}
+	}
+	free (r2r_path);
+	return found;
+#else
+#	warning r2r_chdir not yet supported on Windows
```

Comment:
I don't think we need a warning, I doubt this will ever be supported on windows.